docs/source/services/timelock_service/paxos.rst

```diff
@@ -0,0 +1,40 @@
+.. _timelock-paxos:
+
+=====
+Paxos
+=====
+
+Paxos is an algorithm for *distributed consensus* - that is, getting servers to agree on a value that some server
+*proposes*, in the presence of communication and server failures. Its guarantees include:
+
+#. Safety: All servers will agree on the same value.
+#. Nontriviality: The value must have been *proposed* by at least one of the servers.
+#. Liveness: If a majority of servers is up and able to communicate, then the protocol can make progress.
+
+If the Timelock Server is configured to use Paxos, it will actually use separate instances of Paxos to agree on the
```

Comment:
Can the timelock server be configured to use something other than Paxos? I'd probably remove this paragraph and instead combine it with the second with a slightly different focus. It's slightly hard to parse what we use Paxos for right now. 

This page could be presented with this kind of outline:

- The Paxos Algorithm: basically your high level description, link to the paper.
- How we use Paxos
  - Leadership election (primarily motivated for lock service)
  - async paxos (motivated for timestamp service)